src/test/java/com/github/cameltooling/lsp/internal/hover/CamelLanguageServerHoverTest.java

```diff
@@ -74,4 +75,43 @@ public void testDontProvideDocumentationOnHoverWhenEndingWithAnd() throws Except
 		assertThat(hover.get()).isNull();
 	}
 
+	@Test
+	public void testProvideParameterDocumentationOnHover() throws Exception {
+		CamelLanguageServer camelLanguageServer = initializeLanguageServer("<from uri=\"file:bla?filter=test\" xmlns=\"http://camel.apache.org/schema/spring\"></from>\n");
+		
+		TextDocumentPositionParams position = new TextDocumentPositionParams(new TextDocumentIdentifier(DUMMY_URI+".xml"), new Position(0, 26));
+		CompletableFuture<Hover> hover = camelLanguageServer.getTextDocumentService().hover(position);
+		
+		assertThat(hover.get().getContents().getLeft().get(0).getLeft()).isEqualTo(FILE_FILTER_DOCUMENTATION);		
+	}
+	
+	@Test
+	public void testProvideParameterDocumentationForUnknownParamOnHover() throws Exception {
+		CamelLanguageServer camelLanguageServer = initializeLanguageServer("<from uri=\"file:bla?test=test\" xmlns=\"http://camel.apache.org/schema/spring\"></from>\n");
+		
+		TextDocumentPositionParams position = new TextDocumentPositionParams(new TextDocumentIdentifier(DUMMY_URI+".xml"), new Position(0, 26));
+		CompletableFuture<Hover> hover = camelLanguageServer.getTextDocumentService().hover(position);
+		
+		assertThat(hover.get().getContents().getLeft().get(0).getLeft()).isEqualTo(OptionParamURIInstance.INVALID_URI_OPTION);		
+	}
+	
+	@Test
+	public void testProvideSyntaxForPathParameterOnHover() throws Exception {
+		CamelLanguageServer camelLanguageServer = initializeLanguageServer("<from uri=\"kafka:fl\" xmlns=\"http://camel.apache.org/schema/spring\"></from>\n");
+		
+		TextDocumentPositionParams position = new TextDocumentPositionParams(new TextDocumentIdentifier(DUMMY_URI+".xml"), new Position(0, 19));
+		CompletableFuture<Hover> hover = camelLanguageServer.getTextDocumentService().hover(position);
+		
+		assertThat(hover.get().getContents().getLeft().get(0).getLeft()).isNotNull();		
+	}
+	
+	@Test
+	public void testProvideSyntaxForEmptyPathParameterOnHover() throws Exception {
+		CamelLanguageServer camelLanguageServer = initializeLanguageServer("<from uri=\"kafka:\" xmlns=\"http://camel.apache.org/schema/spring\"></from>\n");
+		
+		TextDocumentPositionParams position = new TextDocumentPositionParams(new TextDocumentIdentifier(DUMMY_URI+".xml"), new Position(0, 17));
+		CompletableFuture<Hover> hover = camelLanguageServer.getTextDocumentService().hover(position);
+		
+		assertThat(hover.get().getContents().getLeft().get(0).getLeft()).isNotNull();		
```

Comment:
what about checking for the exact value instead of just non-null?

Reply:
that is risky. it just needs a small change in the catalog to break this test then.

Reply:
yes, it is the case for all other tests too.
The other way is to be able to mock/provide our own catalog.
Advantages doesn't break on version upgrade, drawbacks if we have a bad catalog or if the structure evolves we won't detect it.